src/analysis/convex/cone.lean

```diff
@@ -220,6 +220,28 @@ set_like.coe_injective $ preimage_comp.symm
 @[simp] lemma mem_comap {f : E →ₗ[𝕜] F} {S : convex_cone 𝕜 F} {x : E} : x ∈ S.comap f ↔ f x ∈ S :=
 iff.rfl
 
+instance : has_add (convex_cone 𝕜 E) := ⟨ λ K₁ K₂,
+{ carrier := { c | ∃ a b, a ∈ K₁ ∧ b ∈ K₂ ∧ a + b = c },
```

Comment:
```suggestion
{ carrier := (K₁ : set E) + (K₂ : set E),
```
maybe it even works without the explicit coercion:
```suggestion
{ carrier := K₁ + K₂,
```

Reply:
and change the the one lemma below accordingly.

Reply:
Yeps, it worked without the explicit coercion.

Reply:
This is causing problems as `set.has_add` is a `def` and not an `instance`. The linter is throwing the following error:

/- The `check_reducibility` linter reports: -/
/- THE FOLLOWING INSTANCES MIGHT NOT REDUCE.
These instances contain one or more declarations that are not instances and are also not marked
`@[reducible]`. This means that type-class inference cannot unfold these declarations, which might mean that type-class inference cannot infer that two instances are definitionally equal. This can cause unexpected errors when this class occurs as an *argument* to a type-class problem. See note [reducible non-instances]. -/
-- analysis/convex/cone.lean
#check @convex_cone.has_add /- This instance contains the declarations [set.has_add], which are semireducible non-instances. -/
Error: convex_cone.has_add - This instance contains the declarations [set.has_add], which are semireducible non-instances.

Reply:
I switched back to the complete definition for now...

Reply:
Actually, I'm trying out adding `@[reducible]` to `set.has_add`. Will only know if this works after the CI is done as the linting error is not caught using `#lint`

Reply:
Using `@[reducible]` isn't working, build errors. Reverting to using the explicit definition.